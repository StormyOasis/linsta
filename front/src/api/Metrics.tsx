//TODO: Client side metrics

export class Metrics {
    private static instance:Metrics | null = null;
    private metrics:any = null;

    private constructor(args: any) {

    }

    public static getInstance() : Metrics {
        if(!Metrics.instance) {
            Metrics.instance = new Metrics({
                host: "localhost",
                port: 8125,                
            });
        }
        return Metrics.instance;
    }

    public increment(key: string) {        
        if(this.metrics == null) {
            throw new Error("Metrics failure");
        }
        this.metrics.increment(key);
    }
}

export default Metrics.getInstance();