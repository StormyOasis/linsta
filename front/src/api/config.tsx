export const HOST ="http://localhost:8080";
export const DEFAULT_PFP = `${HOST}/public/images/profile-user-default-pfp.svg`;