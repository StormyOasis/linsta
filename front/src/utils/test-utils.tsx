import React, { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { RootState } from "../Components/Redux/redux";
import { authInitialState } from '../Components/Redux/slices/auth.slice';
import { modalInitialState } from '../Components/Redux/slices/modals.slice';
import { profileInitialState } from '../Components/Redux/slices/profile.slice';
import { miscInitialState } from '../Components/Redux/slices/misc.slice';
import { configureStore } from '@reduxjs/toolkit';

import Theme from '../Components/Themes/Theme';

export const mockInitialState = (overrides?: Partial<RootState>): RootState => ({
    auth: { ...authInitialState, ...(overrides?.auth ?? {}) },
    modal: { ...modalInitialState, ...(overrides?.modal ?? {}) },
    profile: { ...profileInitialState, ...(overrides?.profile ?? {}) },
    misc: { ...miscInitialState, ...(overrides?.misc ?? {}) },
});

type ExtendedRenderOptions = {
    store?: ReturnType<typeof configureStore>;
    preloadedState?: any;
    route?: string; // ‚Üê allow overriding the initial route
} & Omit<RenderOptions, "queries">;

const dummyAuthReducer = (state = { user: null, isLoading: false, error: null }, _action: any) => state;
const dummyModalReducer = (state = { isOpen: false }, _action: any) => state;
const dummyProfileReducer = (state = { name: '', email: '' }, _action: any) => state;
const dummyMiscReducer = (state = { someData: null }, _action: any) => state;

const renderWithStore = (
    ui: ReactElement,
    {
        preloadedState,
        store = configureStore({
            reducer: {
                auth: dummyAuthReducer,
                modal: dummyModalReducer,
                profile: dummyProfileReducer,
                misc: dummyMiscReducer,
            },
            preloadedState,
        }),
        route = "/",
        ...renderOptions
    }: ExtendedRenderOptions = {}
) => {
    const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
        return (
            <Provider store={store}>
                <MemoryRouter initialEntries={[route]}>
                    <Theme>{children}</Theme>
                </MemoryRouter>
            </Provider>
        );
    };

    return {
        ...render(ui, { wrapper: Wrapper, ...renderOptions }),
        store,
    };
};

export * from "@testing-library/react";
export { renderWithStore };
export type { ExtendedRenderOptions };