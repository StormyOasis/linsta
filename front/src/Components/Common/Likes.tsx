import React from "react";
import styled from "styled-components";

import { Post } from "../../api/types";
import { HOST } from "../../api/config";
import { BoldLink, Div, Span } from "./CombinedStyling";

import HeartSVG from "/public/images/heart.svg";
import HeartFilledSVG from "/public/images/heart-fill.svg";

type ViewLikesTextProps = {
    post: Post;
    handleClick: (post: Post) => void;
}

export const ViewLikesText = (props: ViewLikesTextProps) => {
    if (props.post?.global?.likes == null || props.post.global.likesDisabled || props.post.global.likes.length === 0) {
        return null;
    }

    return (
        <Div>
            <Span>Liked by <BoldLink role="link" href={`${HOST}/${props.post.global.likes[0].userName}`}>{props.post.global.likes[0].userName}</BoldLink>
                {props.post.global.likes.length > 1 && <Span> and <BoldLink role="link" onClick={() => props.handleClick(props.post)}>others</BoldLink></Span>}
            </Span>
        </Div>
    );
}

const Container = styled(Div) <{ $isLiked?: boolean, $width: string, $height: string, $offsetIndex: number }>`
    width: ${props => props.$width};
    height: ${props => props.$height};
    color: ${props => props.$isLiked ? "red" : "black"};
    transform: ${props => `translateX(${25 * props.$offsetIndex}px)`}; 

    &:hover {
        color: ${props => props.theme["colors"].borderDarkColor};
    }
`;

type LikeTogglerProps = {
    isLiked: boolean;
    width?: string;
    height?: string;
    offsetIndex?: number;
    handleClick: any;
}

export const LikeToggler = (props: LikeTogglerProps) => {
    const { isLiked = false, width = "28px", height = "28px", offsetIndex = 0 } = props;

    return (
        <Container
            $offsetIndex={offsetIndex}
            $width={width}
            $height={height}
            $isLiked={isLiked}
            onClick={props.handleClick}>

            {isLiked ?
                <HeartFilledSVG style={{ width, height }} /> :
                <HeartSVG style={{ width, height }} />
            }
        </Container>
    );
};