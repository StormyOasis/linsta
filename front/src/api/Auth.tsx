import { postLogin } from "./ServiceController";

export type AuthUser = {
    token: string;
    id: string;
    userName: string;
}

export const authHeader = () => {
    const headers:any = {};

    // Authentication and Authorization via JWT
    const user = JSON.parse(localStorage.getItem('user') as string) as AuthUser;
    if (user == null) {
        return headers;
    }

    if (user && user.token) {
        headers['Authorization'] = `Bearer ${user.token}`;
    }

    return headers;
}

export const login = async (userName: string, password: string) => {    
    if (userName.trim().length === 0 || password.trim().length === 0) {
        throw new Error("Invalid username or password");
    }

    const result = await postLogin({ userName, password });

    if (result == null || result.status != 200 || result.data == null || result.data.token == null) {
        throw new Error("Invalid username or password");
    }

    return result.data;
}

export const logout = () => {
    localStorage.removeItem("user");
    localStorage.removeItem("recentSearches");
}

export const getCurrentUser = () => {
    const user = localStorage.getItem('user');
    if (user == null) {
        return null;
    }
    return JSON.parse(user);
}