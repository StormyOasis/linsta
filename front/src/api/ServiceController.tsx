import axios from "axios";
import { authHeader } from "./Auth";

let host = "http://localhost:3001"; //TODO: From config or env

export type ServiceResponse = {
    data: any;
    status: number;
    statusText: string;
}

export const postAccountsAttempt = async (data: any): Promise<ServiceResponse> => {
    const res = await axios.post(`${host}/api/v1/accounts/attempt/`, data);
    return {
        data: res.data,
        status: res.status,
        statusText: res.statusText,
    }
};

export const getAccountsCheckUserUnique = async (value: string | number): Promise<ServiceResponse> => {
    const res = await axios.get(`${host}/api/v1/accounts/check/${value}`);
    return {
        data: res.data,
        status: res.status,
        statusText: res.statusText
    }
};

export const getAccountsSendVerifyNotification = async (userData: string): Promise<ServiceResponse>  => {
    const res = await axios.get(`${host}/api/v1/accounts/send_confirm_code?user=${userData}`);
    return {
        data: res.data,
        status: res.status,
        statusText: res.statusText
    }
}

export const postLogin = async (data: any): Promise<ServiceResponse> => {
    const res = await axios.post(`${host}/api/v1/accounts/login`, data);
    return {
        data: res.data,
        status: res.status,
        statusText: res.statusText,
    }
}

export const postForgotPassword = async (data: any): Promise<ServiceResponse> => {
    const res = await axios.post(`${host}/api/v1/accounts/forgot`, data);
    return {
        data: res.data,
        status: res.status,
        statusText: res.statusText,
    }
}

export const postChangePassword = async (data: any): Promise<ServiceResponse> => {
    const res = await axios.post(`${host}/api/v1/accounts/change_password`, data);
    return {
        data: res.data,
        status: res.status,
        statusText: res.statusText,
    }
}