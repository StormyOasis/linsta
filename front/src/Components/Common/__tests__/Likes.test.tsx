import React from "react";
import { screen, fireEvent } from "@testing-library/react";
import { ViewLikesText, LikeToggler } from "../Likes";
import { HOST } from "../../../api/config";
import '@testing-library/jest-dom';
import { renderWithStore } from '../../../utils/test-utils';

jest.mock("../../../api/config", () => ({
    HOST: "http://localhost"
}));

jest.mock("../Icon", () => ({
    HeartFilledSVG: (props: any) => <svg data-testid="heart-filled" {...props} />,
    HeartSVG: (props: any) => <svg data-testid="heart-outline" {...props} />,
}));

describe("ViewLikesText", () => {
    const basePost = {
        user: { userId: "1", userName: "alice" },
        global: {
            likes: [{ userName: "bob", userId: "2" }],
            likesDisabled: false
        }
    };

    it("renders null if no likes", () => {
        const post = { ...basePost, global: { ...basePost.global, likes: [] } };
        const r = renderWithStore(<ViewLikesText post={post as any} handleClick={jest.fn()} />);
        expect(r).toMatchSnapshot();
    });

    it("renders null if likesDisabled and not post owner", () => {
        const post = { ...basePost, global: { ...basePost.global, likesDisabled: true } };
        const r = renderWithStore(<ViewLikesText post={post as any} authUserId="not-owner" handleClick={jest.fn()} />);
        expect(r).toMatchSnapshot();
    });

    it("renders likes text with single liker", () => {
        const r = renderWithStore(<ViewLikesText post={basePost as any} handleClick={jest.fn()} />);
        expect(screen.getByText(/Liked by/i)).toBeInTheDocument();
        expect(screen.getByRole("link", { name: "bob" })).toHaveAttribute("href", `${HOST}/bob`);
        expect(screen.queryByText(/and/)).not.toBeInTheDocument();
        expect(r).toMatchSnapshot();
    });

    it("renders likes text with multiple likers and calls handleClick", () => {
        const handleClick = jest.fn();
        const post = {
            ...basePost,
            global: {
                ...basePost.global,
                likes: [
                    { userName: "bob", userId: "2" },
                    { userName: "carol", userId: "3" }
                ]
            }
        };
        const r = renderWithStore(<ViewLikesText post={post as any} handleClick={handleClick} />);
        expect(screen.getByText(/Liked by/i)).toBeInTheDocument();
        expect(screen.getByText(/and/)).toBeInTheDocument();
        const othersLink = screen.getByText("others");
        fireEvent.click(othersLink);
        expect(handleClick).toHaveBeenCalledWith(post);
        expect(r).toMatchSnapshot();
    });
});

describe("LikeToggler", () => {
    it("renders HeartFilledSVG when isLiked is true", () => {
        const r = renderWithStore(<LikeToggler isLiked={true} handleClick={jest.fn()} />);
        expect(screen.getByTestId("heart-filled")).toBeInTheDocument();
        expect(screen.queryByTestId("heart-outline")).not.toBeInTheDocument();
        expect(r).toMatchSnapshot();
    });

    it("renders HeartSVG when isLiked is false", () => {
        const r = renderWithStore(<LikeToggler isLiked={false} handleClick={jest.fn()} />);
        expect(screen.getByTestId("heart-outline")).toBeInTheDocument();
        expect(screen.queryByTestId("heart-filled")).not.toBeInTheDocument();
        expect(r).toMatchSnapshot();
    });

    it("calls handleClick when clicked", () => {
        const handleClick = jest.fn();
        const r = renderWithStore(<LikeToggler isLiked={false} handleClick={handleClick} />);
        fireEvent.click(screen.getByTestId("heart-outline").parentElement!);
        expect(handleClick).toHaveBeenCalled();
        expect(r).toMatchSnapshot();
    });

    it("applies custom width, height, and offsetIndex", () => {
        const r = renderWithStore(<LikeToggler isLiked={false} width="40px" height="50px" offsetIndex={2} handleClick={jest.fn()} />);
        const container = screen.getByTestId("heart-outline").parentElement!;
        expect(container).toHaveStyle("width: 40px");
        expect(container).toHaveStyle("height: 50px");
        expect(container).toHaveStyle("transform: translateX(50px)");
        expect(r).toMatchSnapshot();
    });
});