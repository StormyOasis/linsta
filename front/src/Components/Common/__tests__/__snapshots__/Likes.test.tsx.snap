// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LikeToggler applies custom width, height, and offsetIndex 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 Likes__Container-sc-1tsl058-0 bKrqWI bQZXYN"
      >
        <svg
          data-testid="heart-outline"
          height="50px"
          stroke="none"
          width="40px"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 Likes__Container-sc-1tsl058-0 bKrqWI bQZXYN"
    >
      <svg
        data-testid="heart-outline"
        height="50px"
        stroke="none"
        width="40px"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`LikeToggler calls handleClick when clicked 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 Likes__Container-sc-1tsl058-0 gzBfFh bjyGil"
      >
        <svg
          data-testid="heart-outline"
          height="28px"
          stroke="none"
          width="28px"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 Likes__Container-sc-1tsl058-0 gzBfFh bjyGil"
    >
      <svg
        data-testid="heart-outline"
        height="28px"
        stroke="none"
        width="28px"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`LikeToggler renders HeartFilledSVG when isLiked is true 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 Likes__Container-sc-1tsl058-0 gzBfFh ecxOSp"
      >
        <svg
          data-testid="heart-filled"
          height="28px"
          stroke="none"
          width="28px"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 Likes__Container-sc-1tsl058-0 gzBfFh ecxOSp"
    >
      <svg
        data-testid="heart-filled"
        height="28px"
        stroke="none"
        width="28px"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`LikeToggler renders HeartSVG when isLiked is false 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 Likes__Container-sc-1tsl058-0 gzBfFh bjyGil"
      >
        <svg
          data-testid="heart-outline"
          height="28px"
          stroke="none"
          width="28px"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 Likes__Container-sc-1tsl058-0 gzBfFh bjyGil"
    >
      <svg
        data-testid="heart-outline"
        height="28px"
        stroke="none"
        width="28px"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`ViewLikesText renders likes text with multiple likers and calls handleClick 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
      >
        <span
          class="CombinedStyling__Span-sc-c3jdxk-1 kSRMWv"
        >
          Liked by 
          <a
            class="CombinedStyling__Link-sc-c3jdxk-7 CombinedStyling__BoldLink-sc-c3jdxk-9 rKNtR bPYExg"
            href="http://localhost/bob"
            role="link"
          >
            bob
          </a>
          <span
            class="CombinedStyling__Span-sc-c3jdxk-1 kSRMWv"
          >
             and 
            <a
              class="CombinedStyling__Link-sc-c3jdxk-7 CombinedStyling__BoldLink-sc-c3jdxk-9 rKNtR bPYExg"
              role="link"
            >
              others
            </a>
          </span>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
    >
      <span
        class="CombinedStyling__Span-sc-c3jdxk-1 kSRMWv"
      >
        Liked by 
        <a
          class="CombinedStyling__Link-sc-c3jdxk-7 CombinedStyling__BoldLink-sc-c3jdxk-9 rKNtR bPYExg"
          href="http://localhost/bob"
          role="link"
        >
          bob
        </a>
        <span
          class="CombinedStyling__Span-sc-c3jdxk-1 kSRMWv"
        >
           and 
          <a
            class="CombinedStyling__Link-sc-c3jdxk-7 CombinedStyling__BoldLink-sc-c3jdxk-9 rKNtR bPYExg"
            role="link"
          >
            others
          </a>
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`ViewLikesText renders likes text with single liker 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
      >
        <span
          class="CombinedStyling__Span-sc-c3jdxk-1 kSRMWv"
        >
          Liked by 
          <a
            class="CombinedStyling__Link-sc-c3jdxk-7 CombinedStyling__BoldLink-sc-c3jdxk-9 rKNtR bPYExg"
            href="http://localhost/bob"
            role="link"
          >
            bob
          </a>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="CombinedStyling__Div-sc-c3jdxk-0 escqfF"
    >
      <span
        class="CombinedStyling__Span-sc-c3jdxk-1 kSRMWv"
      >
        Liked by 
        <a
          class="CombinedStyling__Link-sc-c3jdxk-7 CombinedStyling__BoldLink-sc-c3jdxk-9 rKNtR bPYExg"
          href="http://localhost/bob"
          role="link"
        >
          bob
        </a>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`ViewLikesText renders null if likesDisabled and not post owner 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;

exports[`ViewLikesText renders null if no likes 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
